### ISSUE TITLE
False positive on array destructuring of object that implements ArrayAccess

### ISSUE THREAD

**Issue by GroxExMachine** [kalessil/phpinspectionsea#2007]
| Subject        | Details                                                                       |
| :------------- | :---------------------------------------------------------------------------- |
| Plugin         | Php Inspections (EA Extended), version 6.1 and previous |
| Language level |PHP 8.3                         |

Current behaviour
--
When destructuring an object that implements ArrayAccess into an array we are getting false positive detection:
_Error: [EA] This assignment doesn't make any sense here, as the assigned value isn't an array._ 

In this example, class Example implements ArrayAccess, Countable, IteratorAggregate. This class is a part of a testing framework Codeception.

```php
$example = new Example([1, 2]);
[$a, $b] = $example;
```

<img width="266" height="83" alt="Image" src="https://github.com/user-attachments/assets/25950fde-b1e1-4fd5-a85e-99fc2ee657f1" />

Expected behaviour
--
No error or warning.

Environment details
---
Linux Mint 22.3
PHP 8.3. 
PhpStorm 2025.3.3
Build #PS-253.31033.138, built on February 19, 2026
Runtime version: 21.0.10+7-b1163.108 amd64 (JCEF 137.0.17)
VM: OpenJDK 64-Bit Server VM by JetBrains s.r.o.

**Comment by ea-inspections-team** [comment]
Thank you for reporting @GroxExMachine!
The false positive has been resolved.
